for-upstream/common.nix

```diff
@@ -59,42 +80,33 @@ in
       '';
       type = types.path;
     };
-    binaryCachesPath = mkOption {
-      description = ''
-        Important: Avoid putting secrets in the Nix store. Use a string file
-        location here and deploy the actual file to that location separately.
-
-        JSON file with secrets that authorize the agent to use binary caches.
-        For more, see https://docs.hercules-ci.com/#binaryCachesPath
-      '';
-      type = types.nullOr types.path;
-      default = null;
-    };
     concurrentTasks = mkOption {
       description = "Number of tasks to perform simultaneously, such as evaluations, derivations";
       type = types.int;
       default = 4;
     };
-    package = mkOption {
-      description = "Package containing the bin/hercules-ci-agent program";
-      type = types.package;
-      default = pkgs.hercules-ci-agent;
-      defaultText = "pkgs.hercules-ci-agent";
-    };
-    extraOptions = mkOption {
-      description = ''
-        This lets you can add extra options to the agent's config file, in case
-        you are using an upstreamed module with a newer version of the package.
 
-        These will override the other options in this module.
-
-        We recommend that you use the other options where possible, because
-        extraOptions can not provide a merge function for the contents of the
-        fields.
-        '';
+    /*
+      Internal and/or computed values
+     */
+    finalConfig = mkOption {
```

Comment:
it's quite confusing as this "shadows" previous `finalConfig`.

Reply:
Why couldn't it be just part of `config`?

Reply:
I've 'simply' exposed what used to be just a let binding. I'll flip it around so it follows the usual pattern instead.